enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,504 @@
+[progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
+the command line tool.
+
+For the fully automated workflow, the steps a user might normally
+perform such as approving hosts or triggering the installation will be
+handled by a new client running as a service on node0. The client will
+read the user's inputs from files included in the image and interact
+with the assisted-service REST API on localhost to drive the
+deployment. For the purposes of this document, we will refer to that
+service as a single entity called `create-cluster`, although in
+practice we may divide some of the steps into separate tools.
+
+### Actors in the Workflow
+
+**infrastructure admin** is a human user responsible for infrastructure
+within the data center, such as an image registry, networking, etc.
+
+**cluster creator** is a human user responsible for deploying a cluster.
+
+**orchestration tool** is intended to be a software component provided
+by the end user, but the steps may also be performed by **cluster
+creator** if no software component is available.
+
+**installer** is the OpenShift installer binary, `openshift-install`.
+
+**node0** is the first host to be added to a cluster.
+
+### Steps for Deploying a Disconnected Cluster
+
+1. The infrastructure admin launches an image registry visible to the
+   hosts that will form the cluster and to a host that has internet
+   access so the release image can be downloaded.
+2. The cluster creator obtains the `oc` command line tool and uses `oc`
+   to copy the OpenShift release image into the local image registry
+   in one of the usual ways with [`oc adm
+   mirror`](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-installation-images.html)
+   or [the `oc mirror`
+   plugin](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-disconnected.html).
+3. The cluster creator obtains the OpenShift installer in one of [the
+   usual
+   ways](https://docs.openshift.com/container-platform/4.9/installing/installing_bare_metal/installing-bare-metal.html#installation-obtaining-installer_installing-bare-metal).
+4. The cluster creator triggers their orchestration tool to run the
+   installer.
+5. The orchestration tool prepares an `install-config.yaml`, including
+   `NMState` content to configure the static network settings for the
+   hosts, using input data and templates.
+6. The orchestration tool runs `openshift-install create image`.
+7. The installer extracts a copy of the RHCOS ISO from the OpenShift
```

Comment:
Is the RHCOS ISO currently in the release payload?

Reply:
Yes. That's a relatively new inclusion.